### PR TITLE
Add GH Actions for linting and unit testing

### PR DESCRIPTION
**Pull request by jackofallops**

### CODE REVIEW

**Comment by jackofallops** [comment]
Jobs appear to be executing successfully:
image

File: .github/workflows/depscheck.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+---
+name: Dependency Check
```
Comment (katbyte) [inline .github/workflows/depscheck.yaml:2]:
```suggestion
name: Vendor Dependencies Check
```

File: .github/workflows/lintrest.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+---
+name: Lint rest
```
Comment (katbyte) [inline .github/workflows/lintrest.yaml:2]:
```suggestion
name: GoLang Linting
```

File: .github/workflows/tflint.yaml [hunk 3]
```diff
@@ -0,0 +1,24 @@
+---
+name: Terraform Lint
```
Comment (katbyte) [inline .github/workflows/tflint.yaml:2]:
```suggestion
name: Terraform Schema Linting
```

File: .github/workflows/website-lint.yaml [hunk 4]
```diff
@@ -0,0 +1,22 @@
+---
+name: Website lint
```
Comment (katbyte) [inline .github/workflows/website-lint.yaml:2]:
```suggestion
name: Website Linting
```

File: .github/workflows/website-test.yaml [hunk 5]
```diff
@@ -0,0 +1,22 @@
+---
+name: Website test
```
Comment (katbyte) [inline .github/workflows/website-test.yaml:2]:
```suggestion
name: Website checks
```

**Review by katbyte — APPROVED** [review]
Aside form a few step naming issues LGTM 👍

**Pull request merged**